src/foam/core/reflow/dashboard/DashboardDAOAgents.js

```diff
@@ -0,0 +1,704 @@
+          overflow: 'hidden'
+        }).
+        add(chart).
+      end();
+      
+      // Set block value
+      if (block) {
+        if (block.value) {
+          block.value.value = chart;
+        } else {
+          block.value = chart;
+        }
+      }
+      
+      return chart;
+    },
+    
+    function showPropertyRequiredMessage(e) {
+      e.start('div').
+        style({padding: '20px', textAlign: 'center', color: '$textTertiary'}).
+        add('Please select a property to group by').
+      end();
+    },
+    
+    function convertGroupByToChartData(groupBy, propLabel, chartType) {
+      var data = [];
+      var labels = [];
+      
+      // Extract data from GroupBy sink
+      for ( var key in groupBy.groups ) {
+        if ( groupBy.groups.hasOwnProperty(key) ) {
+          labels.push(key.toString());
+          data.push(groupBy.groups[key].value);
+        }
+      }
+      
+      // Create Chart.js dataset with type-specific styling
+      var dataset = {
+        label: propLabel || 'Count',
+        data: data
+      };
+      
+      // Apply chart-type specific styling with CSS tokens
+      switch(chartType) {
+        case 'bar':
+          dataset.backgroundColor = foam.CSS.returnTokenValue('$primary200', this.cls_, this.__context__);
+          dataset.borderColor = foam.CSS.returnTokenValue('$primary400', this.cls_, this.__context__);
+          dataset.borderWidth = 1;
+          break;
+          
+        case 'pie':
+        case 'donut':
+          var tokenColors = ['$red200', '$blue200', '$yellow200', '$green200', '$purple200', '$orange200'];
+          var colors = tokenColors.map(function(token) {
+            return foam.CSS.returnTokenValue(token, this.cls_, this.__context__);
+          }.bind(this));
+          dataset.backgroundColor = colors.slice(0, data.length);
+          break;
+          
+        case 'line':
+          dataset.backgroundColor = foam.CSS.returnTokenValue('$green100', this.cls_, this.__context__);
+          dataset.borderColor = foam.CSS.returnTokenValue('$green500', this.cls_, this.__context__);
+          dataset.borderWidth = 2;
+          dataset.fill = false;
+          dataset.tension = 0.1;
+          break;
+      }
+      
+      return {
+        labels: labels,
+        datasets: [dataset]
+      };
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.core.reflow.dashboard',
+  name: 'DashboardCountDAOAgent', 
+  extends: 'foam.core.reflow.AbstractDAOAgent',
+  mixins: ['foam.core.reflow.dashboard.DirectChartMixin'],
+
+  requires: [
+    'foam.mlang.sink.Count'
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'label',
+      value: 'Count'
+    }
+  ],
+
+  methods: [
+    function execute(e) {
+      var self = this;
+      
+      // Count the DAO records directly
+      this.dao.select(this.Count.create()).then(function(count) {
+        // Create a simple count display
+        e.start('div').
+          style({
```

Comment:
This isnt how tokens work inside style you will have to do
 `foam.css.returnTokenValue('tokenNam', this.cls_, this.__subContext__)`